MIGRATION.md

```diff
@@ -537,3 +556,652 @@ import { Grid, GridItem } from '@contentful/f36-components';
+/>;
+
+<TextInput
+  id="someInput"
+  name="userEmail"
+  labelText="userEmail"
+  disabled
+  required
+  value="some value"
+  onChange={() => {}}
+/>;
+
+<TextInput
+  id="someInput"
+  name="userEmail"
+  labelText="userEmail"
+  error
+  readOnly
+  width="large"
+  onChange={() => {}}
+/>;
+
+<TextInput
+  id="someInput"
+  name="userEmail"
+  labelText="userEmail"
+  withCopyButton
+  width="small"
+/>;
+
+<Textarea
+  id="someInput"
+  name="userEmail"
+  className="my-extra-class"
+  disabled
+  value="some value"
+/>;
+
+<Textarea
+  id="someInput"
+  name="userEmail"
+  className="my-extra-class"
+  error
+  width="large"
+  value="some value"
+  onChange={() => {}}
+/>;
+```
+
+becomes:
+
+```tsx static=true
+import { CopyButton, TextInput, Textarea } from '@contentful/f36-components';
+
+<TextInput
+  id="someInput"
+  name="userEmail"
+  aria-label="userEmail"
+  className="my-extra-class"
+  defaultValue="some value"
+/>;
+
+<TextInput
+  id="someInput"
+  name="userEmail"
+  aria-label="userEmail"
+  isDisabled
+  isRequired
+  value="some value"
+  onChange={() => {}}
+/>;
+
+<TextInput
+  id="someInput"
+  name="userEmail"
+  aria-label="userEmail"
+  isInvalid
+  isReadOnly
+  onChange={() => {}}
+/>;
+
+<TextInput.Group>
+  <TextInput id="someInput" name="userEmail" aria-label="userEmail" />
+  <CopyButton />
+</TextInput.Group>;
+
+<Textarea
+  id="someInput"
+  name="userEmail"
+  className="my-extra-class"
+  isDisabled
+  defaultValue="some value"
+/>;
+
+<Textarea
+  id="someInput"
+  name="userEmail"
+  className="my-extra-class"
+  isInvalid
+  value="some value"
+  onChange={() => {}}
+/>;
+```
+
+You can also make these changes manually.
+
+### FieldGroup
+
+The `FieldGroup` component was removed, and should be replaced by one of the Layout components which best fits your need. For example:
+
+```tsx static=true
+<FieldGroup>...</FieldGroup>
+```
+
+for keeping the same spacing as before it should become:
+
+```tsx static=true
+<Flex flexDirection="column" gap="spacingXs">
+  ...
+</Flex>
+```
+
+#### How to migrate your FieldGroup components
+
+The migration should be done manually, using the Layout component that best fits the user needs, the layout components are `Box`, `Flex`, `Stack`, and `Grid`.
+
+### Form
+
+In version 4 we removed `spacing` property for the Form component and that's the only API change for this component. For example:
+
+```tsx static=true
+import { Form } from '@contentful/forma-36-react-components';
+
+<Form onSubmit={handleSubmit} spacing="condensed">
+  ...
+</Form>;
+```
+
+will become:
+
+```tsx static=true
+import { Form } from '@contentful/f36-components';
+
+<Form onSubmit={handleSubmit} spacing="condensed">
+  ...
+</Form>;
+```
+
+#### How to migrate your form components
+
+To migrate your v3 `Form` component to the version 4 you can simply run the [codemod](https://github.com/contentful/forma-36/tree/forma-v4/packages/forma-36-codemod) that we prepared for this occasion, like:
```

Comment:
To migrate your `Form` component to v4, run the following [codemod](https://github.com/contentful/forma-36/tree/forma-v4/packages/forma-36-codemod):